test/auth.test.js

```diff
@@ -58,15 +67,48 @@ describe('Auth', () => {
     return client.login('email', 'password')
       .then(userId => expect(userId).toEqual('fake-user-id'));
   });
+});
 
-  it('should return a promise for login with only existing auth data userId', async() => {
-    window.fetch.resolves(mockApiResponse());
-    expect.assertions(1);
+describe('Auth login semantics', () => {
+  const email = 'test_user@domain.com';
+  const password = 'password';
+  let test = new StitchMongoFixture();
+  let th;
+  let client;
 
-    let client = new StitchClient();
-    await client.auth.storage.set(common.USER_AUTH_KEY, mockAuthData());
+  beforeAll(async() => await test.setup());
+  afterAll(async() => await test.teardown());
 
-    return client.login('email', 'password')
-      .then(userId => expect(userId).toEqual('fake-user-id'));
+  beforeEach(async() => {
+    const { apiKey, groupId, serverUrl } = extractTestFixtureDataPoints(test);
+    th = await buildClientTestHarness(apiKey, groupId, serverUrl);
+    await th.configureAnon();
+    client = th.stitchClient;
+  });
+
+  afterEach(async() => await th.cleanup());
+
+  it('should track currently logged in provider type', async() => {
```

Comment:
[minor] Per the ticket, we need a `isAuthenticated` like method to maintain parity with other SDKs. We should have a test that checks a flow of:
1. Verify `isAuthenticated` resolves to false
2. Log in
3. Verify `isAuthenticated` resolves to true

Reply:
`isAuthenticated` is going to be a wrapper over checking `authedId`. It seems weird to have multiple sources of truth for this, but I can add the method.

Reply:
I'd argue that `authedId` should throw if you're not authed.